x-pack/plugin/wildcard/src/main/java/org/elasticsearch/xpack/wildcard/mapper/WildcardFieldMapper.java

```diff
@@ -91,13 +93,98 @@
-            return new TokenStreamComponents(tokenizer);
+            TokenStream tok = new NormaliseThinningFilter(tokenizer);
+            
+            return new TokenStreamComponents(r -> {
+                tokenizer.setReader(r);
+            }, tok);            
+            
+            
         }
     });
+    
+    // @deprecated - used for BWC with elasticsearch 7.9
+    static final NamedAnalyzer WILDCARD_ANALYZER_7_9 = new NamedAnalyzer("_wildcard", AnalyzerScope.GLOBAL, new Analyzer() {
+        @Override
+        public TokenStreamComponents createComponents(String fieldName) {
+            Tokenizer tokenizer = new NGramTokenizer(NGRAM_SIZE, NGRAM_SIZE);
+                return new TokenStreamComponents(tokenizer);
+        }
+    });
+    
+    public static class NormaliseThinningFilter extends TokenFilter {
+        private final CharTermAttribute termAtt = addAttribute(CharTermAttribute.class);
+        
+        /**
+         * Create a new NormaliseThinningFilter, that normalizes token text such that even-numbered ascii values
+         * are made odd and punctuation is replaced with /
+         * 
+         * @param in TokenStream to filter
+         */
+        public NormaliseThinningFilter(TokenStream in) {
+          super(in);
+        }
+        
+        @Override
+        public final boolean incrementToken() throws IOException {
+          if (input.incrementToken()) {
+              normalize(termAtt.buffer(), 0, termAtt.length());
+            return true;
+          } else
+            return false;
+        }
+        
+        public static String normalize(String s) {
+            char[] chars = s.toCharArray();
+            normalize(chars, 0, chars.length);
+            return new String(chars);            
+        }
+        
+        /**
+         * Normalizes a token
+         */
+        public static void normalize(final char[] buffer, final int offset, final int limit) {
+          assert buffer.length >= limit;
+          assert 0 <= offset && offset <= buffer.length;
+          for (int i = offset; i < limit;) {
+            int codepoint = Character.codePointAt(buffer, i, limit);
+            i += Character.toChars(
+                    normalize(codepoint), buffer, i);
+           }
+        }
+
+        private static int normalize(int codepoint) {
+            // Normalize  space ! " # $ % &  ' ( } * + , - . chars to / 
+            if (codepoint >=32 && codepoint <= 47) {
+                return 47; 
+            }
+            // Normalize : ; < = > ? @ chars to / 
+            if (codepoint >=58 && codepoint <= 64) {
+                return 47; 
+            }
+            // Normalize  [ \ ] ^ _ ` chars to / 
+            if (codepoint >=91 && codepoint <= 96) {
+                return 47; 
+            }
+            // Normalize  { | } ~ chars to / 
+            if (codepoint >=123 && codepoint <= 126) {
+                return 47; 
+            }
```

Comment:
Can we generalize the block checks above with `Character.isLetterOrDigit() == false` ?